File: Vanilla/sources/zdoc/media/lua/shared/Library/MetaObject.lua
---@class MetaObject : zombie.iso.MetaObject
---@field type int
---@field x int
---@field y int
---@field def RoomDef
---@field bUsed boolean
MetaObject = {}

---@public
---@param bUsed boolean
---@return void
function MetaObject:setUsed(bUsed) end

---@public
---@return int
function MetaObject:getType() end

---@public
---@return int
function MetaObject:getX() end

---@public
---@return boolean
function MetaObject:getUsed() end

---@public
---@return int
function MetaObject:getY() end

---@public
---@return RoomDef
function MetaObject:getRoom() end

File: Vanilla/sources/zomboid/zombie/iso/MetaObject.java
package zombie.iso;

public final class MetaObject {
   int type;
   int x;
   int y;
   RoomDef def;
   boolean bUsed = false;

   public MetaObject(int var1, int var2, int var3, RoomDef var4) {
      this.type = var1;
      this.x = var2;
      this.y = var3;
      this.def = var4;
   }

   public RoomDef getRoom() {
      return this.def;
   }

   public boolean getUsed() {
      return this.bUsed;
   }

   public void setUsed(boolean var1) {
      this.bUsed = var1;
   }

   public int getX() {
      return this.x;
   }

   public int getY() {
      return this.y;
   }

   public int getType() {
      return this.type;
   }
}
